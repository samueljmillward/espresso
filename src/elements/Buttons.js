import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ButtonBase from '@material-ui/core/ButtonBase';
import Typography from '@material-ui/core/Typography';

const label = [
  {
    url: '/images/beans.jpg',
    title: 'My Beans',
    width: '50%',
  },
  {
    url: '/images/machine.jpg',
    title: 'Pull A Shot',
    width: '50%',
  },
];

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: 300,
    width: '100%',
    height: '100vh',
  },
  image: {
    position: 'relative',
    height: 200,
    [theme.breakpoints.down('xs')]: {
      width: '100% !important', // Overrides inline-style
      height: 100,
    },
    '&:hover, &$focusVisible': {
      zIndex: 1,
      '& $imageBackdrop': {
        opacity: 0.15,
      },
      '& $imageMarked': {
        opacity: 0,
      },
      '& $imageTitle': {
        border: '4px solid currentColor',
      },
    },
  },
  imageButton: {
    color: theme.palette.common.white,
  },
  imageBackdrop: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: theme.palette.common.black,
    opacity: 0.4,
    transition: theme.transitions.create('opacity'),
  },
  imageTitle: {
    position: 'relative',
    padding: `${theme.spacing(2)}px ${theme.spacing(4)}px ${
      theme.spacing(1) + 6
    }px`,
  },
}));

export default function ButtonBases() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {label.map((image) => (
        <ButtonBase
          focusRipple
          key={image.title}
          className={classes.image}
          style={{
            width: image.width,
          }}
        >
          <span className={classes.imageBackdrop} />
          <span className={classes.imageButton}>
            <Typography
              component="span"
              variant="subtitle1"
              color="inherit"
              className={classes.imageTitle}
            >
              {image.title}
            </Typography>
          </span>
        </ButtonBase>
      ))}
    </div>
  );
}
