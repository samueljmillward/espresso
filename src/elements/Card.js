import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

// import beans from '../images/beans.jpg';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
}));

export default function BeanCard() {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Typography
          className={classes.title}
          color="textSecondary"
          gutterBottom
        >
          Roasted on: 18th November.
        </Typography>
        <Typography variant="h5" component="h2">
          House Espresso
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          Ethiopia
          <br />
          caramel, dark chocolate, toffee
        </Typography>
      </CardContent>
      {/* <CardMedia
        component="img"
        style={{ height: '300px' }}
        src={beans}
        title="House Espresso"
      /> */}
      <CardActions>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body2" component="p">
            Grind: 6
          </Typography>
          <Typography className={classes.pos} variant="body2" component="p">
            <br />
            Recipe
            <br />
            Dry: 18g
            <br />
            Final Weight: 36g
          </Typography>
          <Typography variant="body2" component="p">
            Our house espresso is a modern take on a classic. It’s a blend of
            two distinctive coffees, which combine to produce an espresso with a
            deep, rich chocolatey body, pronounced toffee sweetness, a complex
            fruity edge, and a hint of bittersweet cocoa at the finish. It’s
            well balanced and combines beautifully with milk, where it adds
            punchy caramel, smooth milk chocolate notes and gets even sweeter
          </Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
