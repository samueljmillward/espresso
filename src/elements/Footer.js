import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    footer: {
        color: '#c46404',
        backgroundColor: '#642b09',
        paddingTop: '3em',
        width: '100%',
        position: 'absolute',
        bottom: 0,
    },
    col: {
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'row',
        alignItems: 'center',
        margin: 0,
        padding: '.2rem',
    },
}));

const Footer = () => {
    const classes = useStyles();

    return (
        <div className={classes.footer}>
            <div>
                <p className={classes.col}>&copy;{new Date().getFullYear()} Espresso</p>
            </div>
        </div>
    );
};

export default Footer;
