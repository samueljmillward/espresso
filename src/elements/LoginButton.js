import React from 'react';
import styled from 'styled-components';
import { useAuth0 } from '@auth0/auth0-react';

const LoginButton = () => {
  const { loginWithRedirect } = useAuth0();
  return <LoginBtn onClick={() => loginWithRedirect()}>Login</LoginBtn>;
};

const LoginBtn = styled.a`
  padding: 1rem 2rem;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  color: burlywood;
  transition: all 0.3s ease-in;
  font-size: 0.9rem;

  &:hover {
    color: #c46404;
  }
`;

export default LoginButton;
