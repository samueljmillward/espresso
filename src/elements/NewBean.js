import React from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import InputAdornment from '@material-ui/core/InputAdornment';
import FormHelperText from '@material-ui/core/FormHelperText';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import DateFnsUtils from '@date-io/date-fns';
import { format } from 'date-fns';
import {
  MuiPickersUtilsProvider,
  KeyboardDatePicker,
} from '@material-ui/pickers';
import { useForm, Controller } from 'react-hook-form';

import { makeStyles } from '@material-ui/core/styles';

import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from 'yup';

const schema = yup.object().shape({
  name: yup.string().required('please enter a valid name').max(30, 'Too long!'),
  origin: yup.string().required('required field').max(30, 'Too long!'),
  flavours: yup.string().required('required field').max(50, 'Too long!'),
  weight: yup
    .number()
    .required('please enter a valid weight')
    .typeError('must be a number')
    .max(9999, "Sorry, that's too many beans for us!")
    .positive()
    .integer(),
});

const useStyles = makeStyles((theme) => ({
  buttons: {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: '32px',
  },
  errorMessage: {
    color: '#AC3232',
    fontSize: '16px',
  },
  addButton: {
    backgroundColor: '#303f9f',
    marginRight: '5px',
    color: '#fff',
  },
  deleteButton: {
    backgroundColor: '#AC3232',
    color: '#fff',
  },
}));

export const defaultValues = {
  name: 'Old Brown Java',
  origin: 'Brazil',
  weight: 250,
  roastDate: new Date(),
  flavours: 'smokey, brown sugar',
  notes: 'Peak flavour at 1 week',
};

export default function NewBean({ setBeansList }) {
  const classes = useStyles();

  const { register, handleSubmit, control, errors } = useForm({
    resolver: yupResolver(schema),
  });

  const onSubmit = (data) => {
    handleClose();
    console.clear();
    console.log({ data });
    data.roastDate = format(data.roastDate, 'dd-MM-yyyy');
    setBeansList((old) => [...old, data]);
  };

  const [open, setOpen] = React.useState(false);

  const [selectedDate, setSelectedDate] = React.useState(new Date());

  const [values, setValues] = React.useState({
    weight: '',
  });

  const handleDateChange = (date) => {
    setSelectedDate(date);
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const clearData = () => {
    window.localStorage.clear();
    window.location.reload();
  };

  return (
    <div className={classes.buttons}>
      <Button
        className={classes.addButton}
        variant='contained'
        onClick={handleClickOpen}
      >
        Add Bean
      </Button>
      <Button
        className={classes.deleteButton}
        variant='contained'
        onClick={clearData}
      >
        Delete Beans
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='form-dialog-title'
      >
        <DialogTitle id='form-dialog-title'>New Bean</DialogTitle>
        <form onSubmit={handleSubmit(onSubmit)}>
          <DialogContent>
            <TextField
              autoFocus
              margin='dense'
              inputRef={register}
              defaultValue={defaultValues.name}
              name='name'
              id='name'
              label='Name'
              type='text'
              fullWidth
            />
            <p className={classes.errorMessage}>{errors.name?.message}</p>
            <TextField
              autoFocus
              margin='dense'
              inputRef={register}
              defaultValue={defaultValues.origin}
              name='origin'
              id='origin'
              label='Origin'
              type='text'
              fullWidth
            />
            <p className={classes.errorMessage}>{errors.origin?.message}</p>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
              <Grid container justify='space-around'>
                <Controller
                  as={<KeyboardDatePicker />}
                  disableToolbar
                  variant='inline'
                  format='dd/MM/yyyy'
                  margin='normal'
                  name='roastDate'
                  defaultValue={defaultValues.roastDate}
                  control={control}
                  id='date-picker-inline'
                  label='Roast Date:'
                  value={selectedDate}
                  onChange={handleDateChange}
                  fullWidth
                  KeyboardButtonProps={{
                    'aria-label': 'change date',
                  }}
                />
                <TextField
                  autoFocus
                  margin='dense'
                  inputRef={register}
                  name='flavours'
                  defaultValue={defaultValues.flavours}
                  id='flavours'
                  label='Flavours'
                  type='text'
                  fullWidth
                />
                <p className={classes.errorMessage}>
                  {errors.flavours?.message}
                </p>
              </Grid>
            </MuiPickersUtilsProvider>
            <FormHelperText id='outlined-weight-helper-text'>
              Weight
            </FormHelperText>
            <Controller
              as={<OutlinedInput />}
              id='outlined-adornment-weight'
              onChange={handleChange('weight')}
              name='weight'
              control={control}
              defaultValue={defaultValues.weight}
              endAdornment={<InputAdornment position='end'>g</InputAdornment>}
              aria-describedby='outlined-weight-helper-text'
              inputProps={{
                'aria-label': 'weight',
              }}
              labelWidth={0}
            />
            <p className={classes.errorMessage}>{errors.weight?.message}</p>
            <TextField
              className={classes.padding}
              autoFocus
              margin='dense'
              inputRef={register}
              name='notes'
              defaultValue={defaultValues.notes}
              id='notes'
              label='Notes'
              type='text'
              fullWidth
            />
            <DialogActions>
              <Button onClick={handleClose} color='primary'>
                Cancel
              </Button>
              <Button type='submit' color='primary'>
                Submit
              </Button>
            </DialogActions>
          </DialogContent>
        </form>
      </Dialog>
    </div>
  );
}
