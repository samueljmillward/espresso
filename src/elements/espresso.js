// eslint-disable-next-line no-lone-blocks
{
  /* <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body2" component="p">
            Grind: 6
          </Typography>
          <Typography className={classes.pos} variant="body2" component="p">
            <br />
            Recipe
            <br />
            Dry: 18g
            <br />
            Final Weight: 36g
          </Typography>
          <Typography variant="body2" component="p">
            Our house espresso is a modern take on a classic. It’s a blend of
            two distinctive coffees, which combine to produce an espresso with a
            deep, rich chocolatey body, pronounced toffee sweetness, a complex
            fruity edge, and a hint of bittersweet cocoa at the finish. It’s
            well balanced and combines beautifully with milk, where it adds
            punchy caramel, smooth milk chocolate notes and gets even sweeter
          </Typography>
        </CardContent>
      </Collapse> */
}
