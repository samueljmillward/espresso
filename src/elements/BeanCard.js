import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Container from '@material-ui/core/Container';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  container: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
  },
  paddingTop: {
    paddingTop: 0,
  },
}));

export default function BeanCard({ bean }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant='outlined'>
      <CardContent>
        <Typography
          className={classes.title}
          color='textSecondary'
          gutterBottom
        >
          Roast Date: {bean.roastDate}
        </Typography>
        <Typography variant='h5' component='h2'>
          {bean.name}
        </Typography>
        <Typography className={classes.pos} color='textSecondary'>
          {bean.origin}
        </Typography>
        <Container className={classes.container}>
          <Typography color='textSecondary'>{bean.flavours}</Typography>
          <Typography color='textSecondary'>{bean.weight}g</Typography>
        </Container>
      </CardContent>
      <CardActions className={classes.paddingTop}>
        <IconButton aria-label='add to favorites'>
          <FavoriteIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label='show more'
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout='auto' unmountOnExit>
        <CardContent>
          <Typography variant='body1'>Notes:</Typography>
          <Typography variant='body2'>{bean.notes}</Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
