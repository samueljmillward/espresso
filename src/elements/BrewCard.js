import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import Container from '@material-ui/core/Container';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

// import beans from '../images/beans.jpg';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  date: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  container: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
  },
  paddingBottom: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
    paddingBottom: '.8rem',
  },
}));

export default function BrewCard({ brews }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Container className={classes.container}>
          <Typography variant="h5" component="h2">
            {brews.name}
          </Typography>
          <Typography className={classes.date} color="textSecondary">
            Brewed on: {brews.brewDate}
          </Typography>
        </Container>
        <Typography className={classes.paddingBottom} variant="caption">
          Espresso
        </Typography>
        <Container className={classes.paddingBottom}>
          <Typography variant="h6" color="textSecondary">
            Dry Weight: {brews.dryWeight}g
          </Typography>
          <Typography variant="h6" color="textSecondary">
            Grind: {brews.grind}
          </Typography>
        </Container>
        <Container className={classes.container}>
          <Typography variant="h6" color="textSecondary">
            Final Weight: {brews.finalWeight}g
          </Typography>
          <Typography variant="h6" color="textSecondary">
            Ratio: 1:1
          </Typography>
        </Container>
      </CardContent>
      {/* <CardMedia
        component="img"
        style={{ height: '300px' }}
        src={beans}
        title="House Espresso"
      /> */}
      <CardActions className={classes.paddingTop}>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body1">Notes:</Typography>
          <Typography variant="body1">{brews.notes}</Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
