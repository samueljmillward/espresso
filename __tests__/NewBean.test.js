import { defaultValues } from '../src/elements/NewBean';

describe('returns expected data types', () => {
  test('defaultValues match schema', () => {
    const beanSchema = {
      name: expect.any(String),
      origin: expect.any(String),
      weight: expect.any(Number),
      roastDate: expect.any(Object),
      flavours: expect.any(String),
      notes: expect.any(String),
    };
    expect(defaultValues).toEqual(beanSchema);
  });
});
