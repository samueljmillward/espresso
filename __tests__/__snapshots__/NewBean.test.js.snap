// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI NewBean form 1`] = `
<div
  className="makeStyles-buttons-1"
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-addButton-3"
    onClick={[Function]}
    variant="contained"
  >
    Add Bean
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-deleteButton-4"
    onClick={[Function]}
    variant="contained"
  >
    Delete Beans
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="form-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="form-dialog-title"
    >
      New Bean
    </WithStyles(ForwardRef(DialogTitle))>
    <form
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(DialogContent))>
        <WithStyles(ForwardRef(TextField))
          autoFocus={true}
          defaultValue="Old Brown Java"
          fullWidth={true}
          id="name"
          inputRef={[Function]}
          label="Name"
          margin="dense"
          name="name"
          type="text"
        />
        <p
          className="makeStyles-errorMessage-2"
        />
        <WithStyles(ForwardRef(TextField))
          autoFocus={true}
          defaultValue="Brazil"
          fullWidth={true}
          id="origin"
          inputRef={[Function]}
          label="Origin"
          margin="dense"
          name="origin"
          type="text"
        />
        <p
          className="makeStyles-errorMessage-2"
        />
        <MuiPickersUtilsProvider
          utils={[Function]}
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
            justify="space-around"
          >
            <Controller
              KeyboardButtonProps={
                Object {
                  "aria-label": "change date",
                }
              }
              as={
                <PickerWithState
                  allowKeyboardControl={true}
                  invalidDateMessage="Invalid Date Format"
                  maxDate={2100-01-01T00:00:00.000Z}
                  maxDateMessage="Date should not be after maximal date"
                  minDate={1900-01-01T00:00:00.000Z}
                  minDateMessage="Date should not be before minimal date"
                  openTo="date"
                  views={
                    Array [
                      "year",
                      "date",
                    ]
                  }
                />
              }
              control={
                Object {
                  "defaultValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldArrayDefaultValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldArrayNamesRef": Object {
                    "current": Set {},
                  },
                  "fieldArrayValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldsRef": Object {
                    "current": Object {},
                  },
                  "fieldsWithValidationRef": Object {
                    "current": Object {},
                  },
                  "formState": Object {
                    "dirtyFields": Object {},
                    "errors": Object {},
                    "isDirty": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "submitCount": 0,
                    "touched": Object {},
                  },
                  "formStateRef": Object {
                    "current": Object {
                      "dirtyFields": Object {},
                      "errors": Object {},
                      "isDirty": false,
                      "isSubmitSuccessful": false,
                      "isSubmitted": false,
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "submitCount": 0,
                      "touched": Object {},
                    },
                  },
                  "getValues": [Function],
                  "isFormDirty": [Function],
                  "mode": Object {
                    "isOnAll": false,
                    "isOnBlur": false,
                    "isOnChange": false,
                    "isOnSubmit": true,
                    "isOnTouch": false,
                  },
                  "reValidateMode": Object {
                    "isReValidateOnBlur": false,
                    "isReValidateOnChange": true,
                  },
                  "readFormStateRef": Object {
                    "current": Object {
                      "constructor": true,
                      "dirtyFields": true,
                      "errors": true,
                      "isDirty": true,
                      "isSubmitSuccessful": true,
                      "isSubmitted": true,
                      "isSubmitting": true,
                      "isValid": true,
                      "isValidating": true,
                      "submitCount": true,
                      "touched": true,
                    },
                  },
                  "register": [Function],
                  "removeFieldEventListener": [Function],
                  "resetFieldArrayFunctionRef": Object {
                    "current": Object {},
                  },
                  "setValue": [Function],
                  "shallowFieldsStateRef": Object {
                    "current": Object {},
                  },
                  "shouldUnregister": true,
                  "trigger": [Function],
                  "unregister": [Function],
                  "updateFormState": [Function],
                  "updateWatchedValue": [Function],
                  "useWatchFieldsRef": Object {
                    "current": Object {},
                  },
                  "useWatchRenderFunctionsRef": Object {
                    "current": Object {},
                  },
                  "validFieldsRef": Object {
                    "current": Object {},
                  },
                  "validateResolver": [Function],
                  "watchInternal": [Function],
                }
              }
              defaultValue={2021-05-15T00:00:00.000Z}
              disableToolbar={true}
              format="dd/MM/yyyy"
              fullWidth={true}
              id="date-picker-inline"
              label="Roast Date:"
              margin="normal"
              name="roastDate"
              onChange={[Function]}
              value={2021-05-15T00:00:00.000Z}
              variant="inline"
            />
            <WithStyles(ForwardRef(TextField))
              autoFocus={true}
              defaultValue="smokey, brown sugar"
              fullWidth={true}
              id="flavours"
              inputRef={[Function]}
              label="Flavours"
              margin="dense"
              name="flavours"
              type="text"
            />
            <p
              className="makeStyles-errorMessage-2"
            />
          </WithStyles(ForwardRef(Grid))>
        </MuiPickersUtilsProvider>
        <WithStyles(ForwardRef(FormHelperText))
          id="outlined-weight-helper-text"
        >
          Weight
        </WithStyles(ForwardRef(FormHelperText))>
        <Controller
          aria-describedby="outlined-weight-helper-text"
          as={<WithStyles(ForwardRef(OutlinedInput)) />}
          control={
            Object {
              "defaultValuesRef": Object {
                "current": Object {},
              },
              "fieldArrayDefaultValuesRef": Object {
                "current": Object {},
              },
              "fieldArrayNamesRef": Object {
                "current": Set {},
              },
              "fieldArrayValuesRef": Object {
                "current": Object {},
              },
              "fieldsRef": Object {
                "current": Object {},
              },
              "fieldsWithValidationRef": Object {
                "current": Object {},
              },
              "formState": Object {
                "dirtyFields": Object {},
                "errors": Object {},
                "isDirty": false,
                "isSubmitSuccessful": false,
                "isSubmitted": false,
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "submitCount": 0,
                "touched": Object {},
              },
              "formStateRef": Object {
                "current": Object {
                  "dirtyFields": Object {},
                  "errors": Object {},
                  "isDirty": false,
                  "isSubmitSuccessful": false,
                  "isSubmitted": false,
                  "isSubmitting": false,
                  "isValid": false,
                  "isValidating": false,
                  "submitCount": 0,
                  "touched": Object {},
                },
              },
              "getValues": [Function],
              "isFormDirty": [Function],
              "mode": Object {
                "isOnAll": false,
                "isOnBlur": false,
                "isOnChange": false,
                "isOnSubmit": true,
                "isOnTouch": false,
              },
              "reValidateMode": Object {
                "isReValidateOnBlur": false,
                "isReValidateOnChange": true,
              },
              "readFormStateRef": Object {
                "current": Object {
                  "constructor": true,
                  "dirtyFields": true,
                  "errors": true,
                  "isDirty": true,
                  "isSubmitSuccessful": true,
                  "isSubmitted": true,
                  "isSubmitting": true,
                  "isValid": true,
                  "isValidating": true,
                  "submitCount": true,
                  "touched": true,
                },
              },
              "register": [Function],
              "removeFieldEventListener": [Function],
              "resetFieldArrayFunctionRef": Object {
                "current": Object {},
              },
              "setValue": [Function],
              "shallowFieldsStateRef": Object {
                "current": Object {},
              },
              "shouldUnregister": true,
              "trigger": [Function],
              "unregister": [Function],
              "updateFormState": [Function],
              "updateWatchedValue": [Function],
              "useWatchFieldsRef": Object {
                "current": Object {},
              },
              "useWatchRenderFunctionsRef": Object {
                "current": Object {},
              },
              "validFieldsRef": Object {
                "current": Object {},
              },
              "validateResolver": [Function],
              "watchInternal": [Function],
            }
          }
          defaultValue={250}
          endAdornment={
            <WithStyles(ForwardRef(InputAdornment))
              position="end"
            >
              g
            </WithStyles(ForwardRef(InputAdornment))>
          }
          id="outlined-adornment-weight"
          inputProps={
            Object {
              "aria-label": "weight",
            }
          }
          labelWidth={0}
          name="weight"
          onChange={[Function]}
        />
        <p
          className="makeStyles-errorMessage-2"
        />
        <WithStyles(ForwardRef(TextField))
          autoFocus={true}
          defaultValue="Peak flavour at 1 week"
          fullWidth={true}
          id="notes"
          inputRef={[Function]}
          label="Notes"
          margin="dense"
          name="notes"
          type="text"
        />
        <WithStyles(ForwardRef(DialogActions))>
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[Function]}
          >
            Cancel
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            color="primary"
            type="submit"
          >
            Submit
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(DialogActions))>
      </WithStyles(ForwardRef(DialogContent))>
    </form>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
